File: Practice/AOJ/Volumes/0005/volumes0005.rb
def gcd(a, b)
  if a > b
    gcd(b, a)
  end
  # 最大公約数
  d=0
  loop do
    d = b % a
    break if d == 0
    a, b = d, a
  end
  return a
end

def lcm(a, b)
  return (a*b)/(gcd(a, b))
end

loop do
  target = gets
  break if target.nil?
  a, b = target.split.map(&:to_i)
  g = gcd(a, b)
  l = lcm(a, b)
  puts "#{g} #{l}"
end

File: Practice/AOJ/Volumes/0005/volumes0005.py
def gcd(x, y):
    if x > y:
        return gcd(y, x)
    elif x == 0:
        return y
    x, y = y % x, x
    return gcd(x, y)


def lcm(x, y):
    return int((x * y) / gcd(x, y))


# try~except使うのあほくさい なんとかならんか
while True:
    try:
        get = list(map(int, input().split()))
        x, y = get
        print("{} {}".format(gcd(x, y), lcm(x, y)))
    except:
        break
